public/js/dashboard.js

```diff
@@ -13,33 +14,56 @@ const findAncestor = (el, cls) => {
   return el;
 };
 
-const resendEmail = async(e) => {
-  const resendLink = e.target;
-  const parentDiv = findAncestor(resendLink, "e-toggle-info-wrapper");
-  const emailToSend = parentDiv.querySelector("span.unverified-e-address").innerText;
-
-  JSON.stringify(emailToSend);
-
-  await fetch("/user/resend-email", {
+const sendForm = async(action, formBody={}) => {
+  const response = await fetch(`/user/${action}`, {
     headers: {
       "Content-Type": "application/json; charset=utf-8",
     },
     mode: "cors",
     method: "POST",
-    body: JSON.stringify({ "email": emailToSend }),
+    body: JSON.stringify(formBody),
   });
+
+  return await response.json();
+};
+
+const sendCommunicationOption = async(e) => {
+  const radioButton = e.target;
+  const formAction = radioButton.dataset.formAction;
+  const option = radioButton.dataset.commOption;
+  sendForm(formAction, { communicationOption: option })
+    .then(data => {}) /*decide what to do with data */
+    .catch(e => {})/* decide how to handle errors */;
+  };
+
+const removeOrResendEmail = async(e) => {
+  // handles remove email and resend verification link buttons
+  const emailOptionButton = e.target;
+  const emailId =  emailOptionButton.dataset.emailId;
+  const formAction = emailOptionButton.dataset.formAction;
+
+  sendForm(formAction, { emailId: emailId })
+  .then(data => {}) /*decide what to do with data */
+  .catch(e => {})/* decide how to handle errors */;
 };
 
-if (document.querySelector(".toggle-parent")) {
-  const toggles = document.querySelectorAll(".toggle-parent");
+if (document.querySelector(".email-card")) {
+  const toggles = document.querySelectorAll(".toggle");
   toggles.forEach(el => {
     el.addEventListener("click", toggleEl);
   });
 
-  if (document.querySelector(".resend-email")) {
-    const resendLinks = document.querySelectorAll(".resend-email");
-    resendLinks.forEach(link => {
-      link.addEventListener("click", resendEmail);
+  if (document.querySelectorAll(".remove-email")) {
```

Comment:
Can we wrap all this in an immediately-executing anonymous function? To keep it wrapped up away from global context.